CMakeLists.txt

```
@@ -64,3 +65,21 @@ endif()
 if(PROJECT_IS_TOP_LEVEL AND NOT MECH_CONFIG_ENABLE_PYTHON_LIBRARY)
   add_subdirectory(packaging)
 endif()
+
+################################################################################
+# Code Coverage
+if(PROJECT_IS_TOP_LEVEL AND MECH_CONFIG_ENABLE_TESTS)
+  # Test code coverage
+  if(MECH_CONFIG_ENABLE_COVERAGE)
+    include(CodeCoverage)
+    append_coverage_compiler_flags()
+    setup_target_for_coverage_lcov(
+        NAME coverage
+        EXECUTABLE "ctest"
+        EXCLUDE 
+          "${PROJECT_SOURCE_DIR}/test/*"
+          "${PROJECT_SOURCE_DIR}/src/CMakeCXXCompilerId.cpp"
+          "${PROJECT_SOURCE_DIR}/src/CMakeCCompilerId.c"
+        BASE_DIRECTORY "${PROJECT_SOURCE_DIR}/src")
```

```suggestion
    setup_target_for_coverage_lcov(
        NAME coverage
        EXECUTABLE "ctest"
        EXCLUDE 
          "${PROJECT_SOURCE_DIR}/test/*"
        BASE_DIRECTORY "${PROJECT_SOURCE_DIR}/src")
```

Don't think those are needed

These